### PR TITLE
Add a webview to scaffold plugins through creator add subcommand

### PR DESCRIPTION
**Pull request by shatakshiiii**
This PR adds a new webview for adding a plugins(filter, lookup) to an existing collection.

### New `Add Plugin` Webview
 
This view can add plugins (filter, lookup) to an existing collection. The available fields are: `Collection Path`, `Plugin Name` and `Plugin Type`.
- `Collection Path`: The existing collection path that will be the destination where the plugin gets added. If the destination path is an invalid collection, an error message comes up.
- `Plugin Name`: The name you want to create your plugin with.
- `Plugin Type`: The type of plugin user want to add in their collection. Right now ansible-creator has support for adding filter and lookup plugins. The same plugins are supported via vscode-extension webview.

### Webview

image

**JIRA: [AAP-35655](https://issues.redhat.com/browse/AAP-35655)**

### CODE REVIEW

**Comment by abhikdps** [comment]
Can you also add an entry in the codecov.yml file for `addPluginPageApp.ts`? (see https://github.com/ansible/vscode-ansible/pull/1719/files)
This is because webpack mangles the app files to the point where codecov cannot read coverage data.

**Comment by abhikdps** [comment]
@audgirka @alisonlhart Do we also need to remove the log output to file, copy logs and open log file options from this webview just like we removed them from the devfile, devcontainer and the execution-environment webviews?

**Comment by shatakshiiii** [comment]
> Can you also add an entry in the codecov.yml file for `addPluginPageApp.ts`? (see https://github.com/ansible/vscode-ansible/pull/1719/files) This is because webpack mangles the app files to the point where codecov cannot read coverage data.

Done!

**Comment by ssbarnea** [comment]
I still see ci failures

**Comment by alisonlhart** [comment]
> @audgirka @alisonlhart Do we also need to remove the log output to file, copy logs and open log file options from this webview just like we removed them from the devfile, devcontainer and the execution-environment webviews?

Yes, that would be ideal. We can always add them back if it seems needed, but starting with a more streamlined view seems best.

**Comment by audgirka** [comment]
> @audgirka @alisonlhart Do we also need to remove the log output to file, copy logs and open log file options from this webview just like we removed them from the devfile, devcontainer and the execution-environment webviews?

Yes, please remove this option from the UI

**Comment by shatakshiiii** [comment]
@abhikdps @alisonlhart @audgirka Removed the `log output to file`, `copy logs` and `open log file` options from the view!

**Review by tanwigeetika1618 — APPROVED** [review]

File: src/features/contentCreator/addPluginPage.ts [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+
+    let ansibleCreatorAddCommand = await this.getCreatorCommand(
+      pluginName,
+      pluginType,
+      destinationPathUrl,
+    );
+
+    const creatorVersion = await this.getCreatorVersion();
+    if (isOverwritten) {
+      if (semver.gte(creatorVersion, ANSIBLE_CREATOR_VERSION_MIN)) {
```
Comment (abhikdps) [inline src/features/contentCreator/addPluginPage.ts:304]: This is not needed as the add plugins feature got added recently to creator and we are not using the `--force` flag anymore. You can remove this check as well as the else condition where we add the `--force` flag. Instead, add the `--no-overwrite` parameter to the else condition.

**Review by abhikdps — CHANGES_REQUESTED** [review]

File: src/features/contentCreator/addPluginPage.ts [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+            );
+            return;
+        }
+      },
+      undefined,
+      this._disposables,
+    );
+  }
+
+  private async getCreatorVersion(): Promise<string> {
```
Comment (abhikdps) [inline src/features/contentCreator/addPluginPage.ts:245]: This function can also be removed and imported from the utils file if needed anywhere.

File: src/features/contentCreator/addPluginPage.ts (hunk 1 above)
Comment (shatakshiiii) [reply src/features/contentCreator/addPluginPage.ts]: removed checks for `--force` and `ANSIBLE_CREATOR_VERSION_MIN`!

**Review by abhikdps — APPROVED** [review]

File: test/ui-test/contentCreatorUiTest.ts [hunk 3]
```diff
@@ -138,3 +138,71 @@ describe("Test Ansible sample execution environment file scaffolding", () => {
...
+    await overwriteCheckbox.click();
+
+    createButton = await webview.findWebElement(
+      By.xpath("//vscode-button[@id='create-button']"),
+    );
+    expect(createButton, "createButton should not be undefined").not.to.be
+      .undefined;
+
+    expect(
+      await createButton.isEnabled(),
```
Comment (alisonlhart) [inline test/ui-test/contentCreatorUiTest.ts:188]: @shatakshiiii Can you add a test section to unselect overwrite and rescaffold (expected failure) for slightly more test coverage? See this change for reference: https://github.com/ansible/vscode-ansible/pull/1733/files#diff-87cfb0aafd5395ec414531d5cf42578e92cb4995438776ed01207c68c95c6ad0R152-R156

**Review by alisonlhart — DISMISSED** [review]

**Comment by alisonlhart** [comment]
@shatakshiiii Also we can go ahead and sort by title on the sidebar "Add" section like you suggested earlier.

**Comment by shatakshiiii** [comment]
> @shatakshiiii Also we can go ahead and sort by title on the sidebar "Add" section like you suggested earlier.

Sorted!

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -176,6 +176,8 @@ jobs:
 
       - name: Activate direnv
         if: ${{ !contains(matrix.shell, 'wsl') }}
+        env:
```
Comment (alisonlhart) [inline .github/workflows/ci.yaml:179]: This section should be removed now.

File: Taskfile.yml [hunk 5]
```diff
@@ -186,6 +186,8 @@ tasks:
     generates:
       - "*.vsix"
     cmds:
+      - echo $NODE_OPTIONS
+      - export NODE_OPTIONS="--max-old-space-size=8192"
```
Comment (alisonlhart) [inline Taskfile.yml:190]:
```suggestion
```

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -176,6 +176,8 @@ jobs:
...
+          NODE_OPTIONS: "--max_old_space_size=8192"
```
Comment (alisonlhart) [inline .github/workflows/ci.yaml:180]:
```suggestion
```

Comment (alisonlhart) [reply .github/workflows/ci.yaml:180]: Can be removed now.

**Pull request merged**